### PR TITLE
Deadline: Fix Submit Publish Job with Extend Frames enabled

### PR DESCRIPTION
**Pull request by BigRoy**
## Brief description

Extend frames logic is broken due to call to `get_representations` with `extensions` argument (which the function does not have). 
Also removes `get_resource_files` since the function was unused.

This tries to resolve that and in the meantime also does a draft refactor for the readability of the code.

## Description

This is just a **DRAFT** for now to remind us of the fact that Extend Frames functionality is broken to begin with and should get fixed **and tested**.

I'm opening up this PR early with this quick draft refactor to raise a discussion whether it's worth fixing this functionality to begin with. Is it used? Why or why not? Is there another feature to replace it, etc.?

Let me know how to proceed.

## Testing notes:

_To be added when more details are clear for this PR_

### CODE REVIEW

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 1]
```diff
@@ -351,66 +319,68 @@ def _copy_extend_frames(self, instance, representation):
...
+
+        # now we need to translate published names from representation
+        # back. This is tricky, right now we'll just use same naming as
+        # the regular expected files for this render with the source's frame
+        # number
+        staging = representation.get("stagingDir")
+        staging = self.anatomy.fill_roots(staging)
+        copy_transfers = []
+
+        for frame, source in zip(res_collection.indexes, res_collection):
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py:366]: Thank you lord fluffy hound 🐕‍🦺 - I'm aware but I am keeping this around for the "TODO" I've prepared for myself down below.

[the remark this answers is not in the document]
Comment (jakubjezek001) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py:366]: You should not underestimate the powers of mighty Hound ;)

**Comment by Minkiu** [comment]
Friendly ping @BigRoy , I checked the develop branch and this doesn't seem to be there, so safe to assume this PR is still valid?

**Comment by mkolar** [comment]
I think we were actually rebuilding that functionality while switching to the new way of publishing to deadline @antirotor you might remember.

**Comment by mkolar** [comment]
Because we're splitting OpenPype into ayon-core and individual host addons, this PR would have to be re-created to target one of those.

We're closing it down, but we'll he happy for a new PR to [ynput/ayon-core](https://github.com/ynput/ayon-core) or the host addon repository once it's up.

**Pull request closed without merging**